src/lib/storage/vector_compression/bitpacking/bitpacking_compressor.cpp

```diff
@@ -0,0 +1,46 @@
+#include "bitpacking_compressor.hpp"
+
+#include <algorithm>
+#include <cmath>
+
+#include "bitpacking_vector_type.hpp"
+
+namespace opossum {
+
+class BitPackingVector;
+
+std::unique_ptr<const BaseCompressedVector> BitPackingCompressor::compress(const pmr_vector<uint32_t>& vector,
+                                                                           const PolymorphicAllocator<size_t>& alloc,
+                                                                           const UncompressedVectorInfo& meta_info) {
+  const auto max_element_it = std::max_element(vector.cbegin(), vector.cend());
+
+  auto required_bits = 1u;
+  if (max_element_it != vector.cend() && *max_element_it != 0) {
+    // add 1 to max_value because log2(1) = 0 but we need one bit to represent it.
+    required_bits = static_cast<uint32_t>(std::ceil(log2(*max_element_it + 1u)));
+  }
+
+  auto data = pmr_compact_vector(required_bits, vector.size(), alloc);
+
+  /**
+   * The compact::vector does not zero initialize its memory, which leads to non-reproducible tests that
+   * use the binary writer. The compact::vector allocates memory aligned to its word size and due to this,
```

Comment:
Maybe add something like "[..] use the binary writer (which checks written bytes one by one, writing uninitialized bytes at the end of the vector thus leads to "random" bytes written to disk)"?

Reply:
added